elastic/kibana
# [Response Ops] Calculating overdue metrics using aggregation instead of fetch

Resolves https://github.com/elastic/kibana/issues/134956

## Summary

* Exposes `taskStore.aggregate` via task manager plugin API
* Allowing runtime mappings when aggregating
* Updating `registerClusterCollector` for alerting and actions to use aggregation over task manager documents to calculate the overdue task percentile.
  * Previously, the command fetched the default size of 10 documents and then manually calculated the overdue time in milliseconds, and then the `p50` and `p99` values
  * This PR uses runtime mappings to calculate the overdue time in milliseconds and then uses aggregations to generate the percentile value over all task documents that are overdue.

## To Verify

* Create > 10 rules with actions. I created them to run fairly frequently.
* Let the rules run and navigate to `https://localhost:5601/api/monitoring_collection/cluster_rules`. You should see 0 overdue rule tasks and p50/p99 of 0 (depending on how you set up your rules and how many you're running, you may see 1 or 2 overdue tasks)
* Add `xpack.task_manager.unsafe.exclude_task_types: ['alerting:*']` to your `kibana.yml` to stop running all rules
* Refresh `https://localhost:5601/api/monitoring_collection/cluster_rules`. You should see the number of overdue tasks match the number of rules you have set up (and be greater than 10) and the p50/p99 increase with each refresh.
* Change `xpack.task_manager.unsafe.exclude_task_types:` to exclude `actions:*`.
* Navigate to `https://localhost:5601/api/monitoring_collection/cluster_actions`. You should see the number of overdue tasks increase with each refresh (as the actions pile up) and the p50/p99 increase with each refresh.

### Checklist
